services/web/server/src/simcore_service_webserver/garbage_collector_core.py

```diff
@@ -423,7 +423,10 @@ async def remove_guest_user_with_all_its_resources(
     """Removes a GUEST user with all its associated projects and S3/MinIO files"""
 
     try:
-        if not await is_user_guest(app, user_id):
+        user_role: Optional[UserRole] = await get_user_role(app, user_id)
+        if user_role is None or user_role > UserRole.GUEST:
```

Comment:
what is the use case where user role is none? you already have an except on UserNotFoundError below...

Reply:
> what is the use case where user role is none? you already have an except on UserNotFoundError below...
good catch! 


> A few comments. Also I thought this PR would fix the deletion of projects as well. but it does not right?

No. It will next. These are the last remains before that.